src/git/github-git/repo-tags.ts

```diff
@@ -0,0 +1,102 @@
+import { execWithOutput } from "../../utils";
+
+export interface GitTag {
+  name: string;
+  ref: string;
+}
+
+/**
+ * Pushes any missing tags to the remote first replacing those tags with lightweight versions.
+ * We replace annotated tags with lightweight ones because we cannot sign annotated tags, but
+ * lightweight tags pointing to signed commits will show up as verified in GitHub.
+ */
+export async function pushTags(cwd?: string) {
+  const onlyLocalTags = await getOnlyLocalTags(cwd);
+
+  await deleteTags(onlyLocalTags, cwd);
+  const createdTags = await createLightweightTags(onlyLocalTags, cwd);
+  await execWithOutput("git", ["push", "origin", "--tags"], { cwd });
```

Comment:
do we need to worry about the issue of rate limiting issue here when pushing multiple tags?
https://github.com/orgs/community/discussions/24353